lint.tsconfig.json

```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
```

Comment:
why have a separate tsconfig for linting? perhaps we can just include src and test in the main tsconfig?

Reply:
Good question, I might be incorrect here.  I was thinking that we would want a ts config for build, and another for lint.
Basically, don't include the tests in the build.

Reply:
Yeh I go back on forth on that one honestly. But actually, perhaps you are right to keep them separate...

Reply:
pros:
- smaller build fewer deps
cons:
- more complicated tsc setup 

I don't have a strong opinion.  Leave them separate for now?

Reply:
yup